### PR TITLE
User history

### PR DESCRIPTION
**Pull request by UGcode101**
hopefully this is correct

### CODE REVIEW

**Comment by JenniferNedland** [comment]
The UserHistory table should hold more specialized information. If you look at the sql script in the jenn/database branch, you can see how the UserHistory holds one entry for each generated tracklist, where each entry contains the name of the tracklist and a reference to the GenreStats table, where the genres are stored. This way the user is tied to their statistics entry, but the statistics entry is not tied back to the user.

The advantage of doing it this way is that the statistics are disconnected from the user, so that if we delete the user object the statistics will stay behind. Also, we can get anonymized statistics from across the entire user base.

Similarly, there's a separate BlockStats table, which holds blocked songs and artists. Since we've divided the blocked entities into songs and artists in the current table, we could do a BlockedSongStats and BlockedArtistStats table. Then the BlockedSongs table can point to the BlockedSongStats entry, and avoid storing the same thing in multiple places, while again getting the benefit of having stored the stats independent of the users, giving us access to anonymized statistics from across the entire user base.

So to sum it up, you should create a GenreStats, a BlockedSongStats, and a BlockedArtistStats entity, and have a one-to-one relation from UserHistory to GenreStats, a one-to-one relation from BlockedSong to BlockedSongStats, and a one-to-one relation from BlockedArtist to BlockedArtistStats.

**Review by JenniferNedland — CHANGES_REQUESTED** [review]
See my other comment.
